merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/PostgresResultsCellRepository.java

```diff
@@ -44,34 +45,44 @@ public PostgresResultsCellRepository(final DataSource dataSource) {
   @Override
   public ResultsProtocol.OwnerRole allocate(final PlanId planId) {
     try (final var connection = this.dataSource.getConnection()) {
-      final var planStart = getPlan(connection, planId).startTime();
-      // TODO: At the time of writing, simulation starts at the plan start every time
-      //       When that changes, we will need to update the simulation start here as well
-      final var simulationStart = planStart;
-      var simulation$ = getSimulation(connection, planId);
-
-      final SimulationRecord simulation;
-      if (simulation$.isPresent()) {
-        simulation = simulation$.get();
-      } else {
-        simulation = createSimulation(connection, planId, Map.of());
+      final SimulationRecord simulation = getSimulation(connection, planId);
+      final SimulationTemplateRecord template;
+      Timestamp startTime = simulation.simulationStartTime();
+      Timestamp endTime = simulation.simulationEndTime();
+      final var arguments = new HashMap<String, SerializedValue>();
+
+      if (startTime == null || endTime == null) {
+        if (simulation.simulationTemplateId().isPresent()) {
+          try (final var getSimulationTemplate = new GetSimulationTemplateAction(connection)) {
+            final var templateOptional = getSimulationTemplate.get(simulation.simulationTemplateId().get());
+            if (templateOptional.isEmpty()) {
+              throw new RuntimeException("TemplateRecord should not be empty");
+            }
+            template = templateOptional.get();
+            startTime = (startTime == null ? template.simulationStartTime() : startTime);
+            endTime = (endTime == null ? template.simulationEndTime() : endTime);
```

Comment:
If there is a template, but the template's start or end time is null - that means `startTime` or `endTime` could be null after line 63, right? Is this a nominal scenario, or will this trigger a null pointer exception later on?

Reply:
Thanks for the catch! So, error-wise, it'll throw an NPE in createSimulationDataset() on line 73 when it tries to parse the timestamp, but I'll change it to a better error.

Reply:
Fixed!